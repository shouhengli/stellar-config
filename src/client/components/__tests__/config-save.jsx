const React = require('react');
const renderer = require('react-test-renderer');
const Save = require('../config-save.jsx');

const {
  CONFIG_STATUS_CHANGED,
  CONFIG_STATUS_SAVING
} = require('../../config-status');

describe('component config-save', () => {
  test('is clickable when config has been changed', () => {
    const props = {
      configType: 'source',
      configName: 'vehicles',
      configContent: 'vehicles: {id: integer, model: string}',
      configStatus: CONFIG_STATUS_CHANGED,
      handleClick: jest.fn()
    };

    const component = renderer.create(<Save {...props} />);
    const tree = component.toJSON();
    tree.props.onClick();

    expect(tree).toMatchSnapshot();
    expect(props.handleClick).toHaveBeenCalledTimes(1);
    expect(props.handleClick).toHaveBeenCalledWith(
      props.configType,
      props.configName,
      props.configContent
    );
  });

  test('shows loading icon when config is being saved', () => {
    const props = { configStatus: CONFIG_STATUS_SAVING };

    const component = renderer.create(<Save {...props} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
