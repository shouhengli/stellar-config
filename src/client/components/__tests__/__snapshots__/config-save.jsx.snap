// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component config-save is clickable when config has been changed 1`] = `
<span
  className="button is-primary"
  onClick={[Function]}
>
  <span
    className="icon"
  >
    <i
      className="fa fa-upload"
    />
  </span>
</span>
`;

exports[`component config-save shows loading icon when config is being saved 1`] = `
<span
  className="button is-loading"
>
  <span
    className="icon"
  >
    <i
      className="fa fa-spinner"
    />
  </span>
</span>
`;
